core/bridges/cache.go

```diff
@@ -239,6 +239,10 @@ func (c *Cache) doBulkUpsert() {
 	values := maps.Values(c.bridgeLastValueCache)
 	c.mu.RUnlock()
 
+	if len(values) == 0 {
+		return
+	}
+
 	if err := c.ORM.BulkUpsertBridgeResponse(context.Background(), values); err != nil {
```

Comment:
Out of scope, but we can do better than Background here. Let's circle back to upgrade the `stop` chan to a `StopChan` in order to spawn a Context to pass here (or from the caller).